Include/pegen_interface.h

```diff
@@ -4,43 +4,49 @@
 extern "C" {
```

Comment:
At this point, we could rename this header to `parser_interfance.h` so the pegen name does not leak to the public stuff. What do you think?

Reply:
Very good point! Done in https://github.com/python/cpython/pull/21016/commits/fd6bbe2a9a20eded451e0d2119e0cd294b8cca86.